source/includes/_node_rpc_mining.md

```diff
@@ -0,0 +1,618 @@
+  "target": "0000000000000000000000000000000000000000000000f0ff0f000000000000",
+  "height": 1178782
+}
+```
+
+Long polling for new work.
+
+Returns new work, whenever new TX is received in the mempool or
+new block has been discovered. So miner can restart mining on new data.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+None. |
+
+
+
+## getblocktemplate
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "getblocktemplate",
+    "params": []
+  }'
+```
+
+```shell--cli
+bcoin cli rpc getblocktemplate
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  const res = await rpc.execute('getblocktemplate');
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+{
+  "capabilities": [
+    "proposal"
+  ],
+  "mutable": [
+    "time",
+    "transactions",
+    "prevblock"
+  ],
+  "version": 536870912,
+  "rules": [
+    "csv",
+    "!segwit"
+  ],
+  "vbavailable": {},
+  "vbrequired": 0,
+  "height": 1178933,
+  "previousblockhash": "0000000000000e19ff0c98e7d99987071a791c47338f71389342f7457eeb8695",
+  "target": "0000000000000ffff00000000000000000000000000000000000000000000000",
+  "bits": "1a0ffff0",
+  "noncerange": "00000000ffffffff",
+  "curtime": 1502974969,
+  "mintime": 1502974260,
+  "maxtime": 1502982169,
+  "expires": 1502982169,
+  "sigoplimit": 80000,
+  "sizelimit": 4000000,
+  "weightlimit": 4000000,
+  "longpollid": "0000000000000e19ff0c98e7d99987071a791c47338f71389342f7457eeb86950000000018",
+  "submitold": false,
+  "coinbaseaux": {
+    "flags": "62636f696e206d6163626f6f6b"
+  },
+  "coinbasevalue": 156465644,
+  "default_witness_commitment": "6a24aa21a9edc9f8a77e851392ce58c6b5c64fafbebd76e92c03a198d1e2c942d71fe05c7bc4",
+  "transactions": [
+    {
+      "data": "010000000194a1dd786bc486412dd4125c90eeac0550131b7ffee190d3c8bce92007dfe682000000006b483045022100e2ee65fcdb3c005e430c279e759764937e70993c3a97c21b8580b16146766cdc022044cbb4a09ce4ed4f9c6be14befb6fbccd7819814073f33220e076ced7b103fca01210396e891cf6f1e71e1a266abd3c1e990331da6dc2d3b41e53c6e991f66262f2f32ffffffff022aee0500000000001976a914a21138d3322b67910f66f676396ea432faee34cb88aca0860100000000001976a91494178f0b288078717873b4228e2aad27b91a492188ac00000000",
+      "txid": "809c3d8fb6524acc4ffcaeb2a97ccb1c4c1124ccc5034745ba6ac43f8d4d63c9",
+      "hash": "809c3d8fb6524acc4ffcaeb2a97ccb1c4c1124ccc5034745ba6ac43f8d4d63c9",
+      "depends": [],
+      "fee": 11350,
+      "sigops": 8,
+      "weight": 904
+    },
+    ...
+  ]
+}
+```
+
+returns block template or proposal for use with mining.
+Also validates proposal if `mode` is specified as `proposal`.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+1  | jsonrequestobject | {} | JSONRequestObject.
```

Comment:
I've not included. This option is for Miners when they want to configure what their capabilities are and what they want to control. There are lots of options and are described in multiple BIPs: BIP22, BIP23, BIP145 (for segwit updates). Because it's that specific to miners I left it out.